commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -322,4 +325,79 @@ private String outOfBoundsMsg(int index) {
         return INDEX_MSG + index + SIZE_MSG + size;
     }
 
+    /**
+     * This class acts as an iterator over the ComplexList class.
+     */
+    private class ComplexCursor implements ComplexSink<Void> {
+
+        /**
+         * Index of element to be returned by subsequent call to next.
+         */
+        private int index;
+
+        /**
+         * The modCount value that the cursor believes that the backing List should have.
+         * If this expectation is violated, the cursor has detected concurrent modification.
+         */
+        private final int expectedModCount = modCount;
+
+        /**
+         * Checks if the iteration has more elements.
+         * @return true if the iteration has more elements.
+         */
+        public boolean next() {
+            return index != size;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public Void apply(double real, double imaginary) {
+            ComplexList.this.realAndImagParts[index * 2] = real;
+            ComplexList.this.realAndImagParts[(index * 2) + 1] = imaginary;
+            return null;
+        }
+
+        /**
+         * Gets the real part \( a \) of the complex number \( (a + i b) \) in the list.
+         *
+         * @return The real part.
+         */
+        public double real() {
+            return ComplexList.this.realAndImagParts[index * 2];
+        }
+
+        /**
+         * Gets the imaginary part \( b \) of the complex number \( (a + i b) \) in the list.
+         *
+         * @return The imaginary part.
+         */
+        public double imag() {
+            return ComplexList.this.realAndImagParts[(index * 2) + 1];
+        }
+
+        /**
+         * Checks for concurrent modification by looking at the modCount.
+         * @throws ConcurrentModificationException if expected modCount isn't equal to modCount.
+         */
+        final void checkForComodification() {
+            if (modCount != expectedModCount) {
+                throw new ConcurrentModificationException();
+            }
+        }
+    }
+
+    /**
+     * Replaces each element of the list with the result of applying the operator to that element.
+     * @param operator The operator to apply to each element.
+     */
+    public void replaceAll(ComplexUnaryOperator<Void> operator) {
+        ComplexCursor cursor = new ComplexCursor();
+        while (cursor.next()) {
```

Comment:
This still advances the index outside of the cursor. I think the cursor should have that responsibility.

The cursor could be changed to use the action in a tryAdvance(action) pattern. Alternatively if the cursor will not be reused then it could be removed and the logic placed directly here. It would eliminate the object allocation and simplify the loop logic.

```Java
final double[] parts = ...;
final int m = size;
for (int i = 0; i < m; i++) {
    final int index = i << 1;
    operator.apply(parts[index], parts[index + 1], (x, y) -> { parts[index] = x; parts[index + 1] = y; });
}
// check for comodification
```
Doing it this way uses a local copy of the list and size for efficiency.